### ISSUE TITLE
geogig sync not working

### ISSUE THREAD

**Issue by smesdaghi** [ROGUE-JCTD/MapLoom#161]
looks like there is an issue with sync. even after making changes so that remotes can be added, despite the 'ping' command working, sync fails. Seems like endTransaction has changed in latest geogig. After a beginTransaction, pull, and a push succeed on two repos on the same commit, the corresponding endTransaction fails:
```
http://192.168.99.100/geoserver/geogig/8f0f74a0-1e7f-46cd-8a07-9297ddf42570/endTransaction?output_format=JSON&_dc=1440802717964&transactionId=31721e79-24c9-4947-9fee-714b550e0fa9&cancel=true

{"response":{"success":false,"error":"org.locationtech.geogig.api.GeogigTransaction.close(GeogigTransaction.java:88)\torg.locationtech.geogig.api.plumbing.TransactionEnd._call(TransactionEnd.java:117)\torg.locationtech.geogig.api.plumbing.TransactionEnd._call(TransactionEnd.java:45)\torg.locationtech.geogig.api.AbstractGeoGigOp.call(AbstractGeoGigOp.java:132)\torg.locationtech.geogig.web.api.commands.EndTransaction.run(EndTransaction.java:66)\t"}}
```

**Comment by smesdaghi** [comment]
The sync produces an error on endTransaction because GeogigTransaction.close, transactionBlobStore is null. a null check fixes the sync successfully but looking to se why the ref is null even though it was created on .create(). Looks like perhaps transaction object was deserialized (or cloned) since it is a different ref and could be that the blobstore needs to be created/copied in the constructor as opposed to create.

**Comment by smesdaghi** [comment]
added pull request: https://github.com/locationtech/geogig/pull/20, one resolved will build new geoserver.war file and update the release.

**Comment by smesdaghi** [comment]
fixed in geoshape release 1.5
